package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -1,16 +1,43 @@
-import numpy as np
 import sys
 import os
 import pyh5md
 import MDAnalysis as mda
-from MDAnalysis.coordinates import base
+from MDAnalysis.coordinates import base, core
 
+class Timestep(base.Timestep):
+    """H5MD Timestep
+    
+    """
+    order = 'C'
+    
+    def _init_unitcell(self):
+        return np.zeros((3, 3), dtype=np.float32)
+    
+    @property
+    def dimensions(self):
+        """unitcell dimensions (*A*, *B*, *C*, *alpha*, *beta*, *gamma*)
+
+        lengths *a*, *b*, *c* are in the MDAnalysis length unit (Å), and
+        angles are in degrees.
+
+        Setting dimensions will populate the underlying native format description (triclinic box vectors)
+        If `edges<https://nongnu.org/h5md/h5md.html#simulation-box>`_ is a matrix, 
+        the box is of triclinic shape with the edge vectors given by the rows of the matrix.) 
+        """
+        return core.triclinic_box(*self._unitcell) # vectors are the rows
+
+    @dimensions.setter
+    def dimensions(self, box):
+        self._unitcell[:] = core.triclinic_vectors(box)
+    
 class H5MDReader(base.ReaderBase):
     """Reader for the H5MD format.
 
     """
     format = 'H5MD'
     units = {'time': None, 'length': None}
```

Comment:
document that currently all data are read directly from the file without unit conversion

Reply:
Please also check the h5md specs to see how units are to be handled. Are units recorded somewhere? We could take the approach currently used in the NCDF (AMBER TRJ.py) reader and fail if the units are not ps and Å. Or we find a way to work with any units, see also #2230 .

Discuss how to handle units in the comments.

Reply:
Yep, will look into this. Thank you for the references